### ISSUE TITLE
Discussion on documenting the project

### ISSUE THREAD

**Issue by Anton-Latukha** [haskell-nix/hnix#693]
I am most directly interested that during our work we can simultaneously structure the project by #555.

That means a couple of words here and there to structure and organize the internal code.

Having Haddock coverage of module interface functions seems also beneficial to do during work on the code, when someone directly works with a function and understood it properly.

**Comment by Anton-Latukha** [comment]
Responding to the https://github.com/haskell-nix/hnix/issues/690#issuecomment-674052132.

Thank you for the honest response.

##### I only tried to explain and discuss

and show that from my experience policing documentation too much is not an effective strategy of documentation, and waited to receive actual counter-arguments, so we can arrive at mutual understanding.

##### Why I write too much?

Because I try to discuss things, and you seem to not responde much, I wait on argumentation, and a dialog, so we arrive at the agreement, because the agreement is not met - I continue to discuss the question until we agree on something, because the least I want is to do work and submit it when our process is not discussed.


##### My experience

Writing articles, tech and scientific, being under review, reviewing articles, writing a book, participating in open source projects in over 10 years or so, participating in Arch Wiki and Wikipedia - I've seen clearly a number of times. It is too easy to criticize the human language text, and people think that is "effective". Over the years I, again and again, observe that because maintainers employ the code review to the documentation without consideration of theoretical differences of applying the same algorithm to the very different goals and nature - good documentation PRs do not merge, and devs seem to think it is a contributor issue and not theirs.


##### It is the most effective to contribute docs during writing code

Most effective to do it during the code writing, you work on some function - and that is the moment when you are the person that understands the function better then others, so you add the Haddock commit into the code PR.

And since PRs with code and doc on it - is the most efficient way, but people employ closed documentation review - that creates a problem not only for any documentation, but also to the submitted related code, - maintainers essentially tend to throw the baby with the bathwater there. That maintainers like to block the PRs on the ground of word choise - is the actually only reason why I submit code PR without any explanation or submit docs on it in separate PR, because code without any descriptions to it - passes the devs review much easier then the code with the doc, on the word choise devs often become picky and not understand that back'N'forth has a big effort costs, forcing me to use the speach the way they want. So there is no documentation, PRs with docs stuck for years easily. So I tell them - I did work for you and your project here and provided it for free - take it or leave it it is up to you, if you want to edit it - do not force it on me, edit it how do you like it - it is the most effective approach - is that is unreasonable?

To draw an analogy - requesting person to edit text the way you like it - is a good manager strategy only when it is hard for the person to do or take responsibility, so they delegate it to others. But lets not forget that also for example, babies can not feed or do stuff themselves and can not take responsibility for it - so they delegate to parents to feed, dress them - but only the way they personally want so ofter scandal errupts - in the process of demanding and not accepting the way the person does it. Maintainers are mostly fully capable to evaluate the quality of the docs, they know do the docs integrate with the project even better then contributor, since maintainer knows the full picture and understanding of the project much better then contributor can. Maintainers are more than capable to edit the English language on the spot, to not drag the contributor that already did the work through PR back'N'forth for 2 years exhausting the contributor and blocking the progress.

I see no technical issues to add a commit that changes the English text.

#### I suggested a number of solutions, them provide a lot of middle ground, I even provided to agree with you, but you did not accept any

1. I am up for Wiki model, since it works the best in the world, it showed it is the best strategy of knowledge we know, and scalable almost infinitely to the size of the community, and since it suites for the open projects and communities. The closed model is Britannica.

We know the history.

It is what "The Cathedral and the Bazaar" is about, docs should be for community use or a fenced sacred garden.

Closed project documentation tends to be not understandable and puzzling and to not explain some important parts - it for reason - doc submission process is closed and corporate, so nobody evolved it enough. Most of time it is too much work to submit any doc when the process is closed.

2. https://github.com/haskell-nix/hnix/issues/675#issuecomment-668484952
> I even more open then you for the process, reviewers can get my text edit it without regard to me, since the groundwork is accepted and done - I trust them improving on it.
> What I am up for, is that you can take charge of result text information, but I would ask to accept my doc contributions openly, and then I trust people (and you also) to edit documentation to your liking.

Since you want the closed doc quality reviews - logically I suggested taking the responsibility to what you want.

You seem to be under the impression that I spam the unneeded swaths of documentation everywhere and all the time, meanwhile, that is not my style, especially for Haskell language which is often self-evident when a person understands the language.

I write texts to find the middle ground, when I did not succeed to reach a person and find middle ground with them - what I can do but to try to talk to them more.

3. I even already gave you option of accepting your process: https://github.com/haskell-nix/hnix/issues/675#issuecomment-671927854
> Since I plan to submit a bunch of docs I think we should arrive at the middle ground on this question.

> If you want it so much - I am Ok with having rigorous peer-review of docs, as I already have shown my points and you would understand both the shortcomings of this paradigm for documentation. If the approach would not work for me or other contributors - me/they can contribute code and not docs, as project and people often accustomed doing it this way.

4. Shown that there is a number of ways for maintainer to do the text editing: https://github.com/haskell-nix/hnix/pull/689#issuecomment-672133270

> Maintainer chooses the most counter-productive approach:
> 
> 1. To just review and nitpick the text, and so force the person to comply with own words to the maintainer liking the way maintainer particularly wants it. Essentially placing central scientific problems of semantics, social psychology and telecommunications - right in the middle of the process.
> 
> I've could:
> 
> 2. Submit my edits into the PR.
> 3. Submit them after the merge.
> 4. Or make a correction Hackage release.
> 
> But maintainer decides to not use all 4 tools for the process flow, to use only the one, the most unproductive one.

I both shown all the options and provided to you all options, but somehow you do not agree with any and state that you thinking about living the project. It is all up to you. The doc question is really of moderate importance, it is a good question to learn for us to agree on, but it is not what projects process is about, so I see no problem that we have differences in views, we have a number of ways: for me to just not submit any documentation, or self-merge it, make edits on the spot, or you can review it - however, you want - I am up to agree to any solution.

P.S.

I really want in the future that we would be able to agree faster and better, so I can stop paying time to write long letters.

**Comment by Anton-Latukha** [comment]
Well, my actions and yupping were really about not what they looked like they were about or what I directly talked about. In reality it was about "radical action to unseat the hold of the editorial mind" in a great person and maintainer of community projects which do commune documentation.
